verilog/analysis/verilog_linter_test.cc

```diff
@@ -439,6 +441,87 @@ TEST(VerilogLinterDocumentationTest, AllRulesMarkdown) {
   EXPECT_TRUE(absl::StrContains(stream.str(), "Enabled by default:"));
 }
 
+TEST(VerilogLinterDocumentationTest, PrintLintRuleFile) {
+  auto config_status = verilog::LinterConfigurationFromFlags("");
+  EXPECT_TRUE(config_status.ok());
+  if (!config_status.ok()) {
+    return;
+  }
+  const LinterConfiguration &config = *config_status;
+
+  // Generate Line Rule File
+  std::ostringstream stream;
+  verilog::GetLintRuleFile(&stream, config);
+
+  // Spot-check a few patterns, must mostly make sure generation
+  // works without any fatal errors.
+  // NOTE: This will break if/when the rules change so this part
+  // of the test is not ideal.
+  EXPECT_TRUE(absl::StrContains(stream.str(), "always-comb"));
+  EXPECT_TRUE(absl::StrContains(
+      stream.str(), "module-filename=allow-dash-for-underscore:false"));
+  EXPECT_TRUE(absl::StrContains(stream.str(), "-forbid-negative-array-dim"));
+
+  // Roundtrip test, first parse the rules
+  std::string generated_default_rules_str = stream.str();
+  absl::StatusOr<std::string> generated_default_rules =
+      generated_default_rules_str;
+  RuleBundle parsed_rules;
+  std::string error;
+  parsed_rules.ParseConfiguration(*generated_default_rules, '\n', &error);
+  EXPECT_TRUE(error.empty());
+
```

Comment:
at this point, you have a `RuleBundle` from parsing.
Wouldn't it just be sufficient to compare that `parsed_rules` is the same as you'd get from `config.GetRuleBundle()` ? Otherwise you're essentially re-creating the parsing logic from `RuleBundle::ParseConfiguration` here, which makes things more brittle.

Reply:
As it stands, config.GetRuleBundle() dosn't inlcude the disabled rules or the default configuration for each rule.

I could pass parsed_rule_bundle back through UnparseConfiguration and compare against the generated rules string. Something like this perhaps?
 
```
  std::string generated_default_rules_str verilog::GetLintRuleFile(&stream, config).str()

  // Roundtrip test, first parse the rules
  parsed_rule_bundle.ParseConfiguration(*generated_default_rules_str, '\n', &error);

  // Convert the parsed_rule_bundle back to a string
  std::string unparsed_rule_bundle = parsed_rule_bundle.UnparseConfiguration('\n', false);

  // compare the unparsed_rule_bundle to the generated_default_rules_str
  EXPECT_TRUE(unparsed_rule_bundle == generated_default_rules_str));
}
```

Reply:
I've pushed some changes across, let me know if there's anything else I should address.